ykman/cli/piv.py

```diff
@@ -223,22 +224,25 @@ def generate_key(
 
     _ensure_authenticated(ctx, controller, pin, management_key)
 
-    algorithm = ALGO.from_string(algorithm)
+    algorithm_id = ALGO.from_string(algorithm)
```

Comment:
Why this name change?

Reply:
Because previously, `algorithm` changed type halfway so the error message would show "Algorithm 6 is not supported on this YubiKey".